protocol/wlr-output-power-management-unstable-v1.xml

```diff
@@ -0,0 +1,128 @@
+
+  <description summary="Control power management modes of outputs">
+    This protocol allows clients to control power management modes
+    of outputs that are currently part of the compositor space. The
+    intent is to allow special clients like desktop shells to power
+    down outputs when the system is idle.
+
+    To modify outputs not currently part of the compositor space see
+    wlr-output-management.
+
+    Warning! The protocol described in this file is experimental and
+    backward incompatible changes may be made. Backward compatible changes
+    may be added together with the corresponding interface version bump.
+    Backward incompatible changes are done by bumping the version number in
+    the protocol and interface names and resetting the interface version.
+    Once the protocol is to be declared stable, the 'z' prefix and the
+    version number in the protocol and interface names are removed and the
+    interface version number is reset.
+  </description>
+
+  <interface name="zwlr_output_power_manager_v1" version="1">
+    <description summary="manager to create per-output power management">
+      This interface is a manager that allows creating per-output power
+      management mode controls.
+    </description>
+
+    <request name="get_output_power">
+      <description summary="get a power management for an output">
+        Create a output power management mode control that can be used to
+        adjust the power management mode for a given output.
+      </description>
+      <arg name="id" type="new_id" interface="zwlr_output_power_v1"/>
+      <arg name="output" type="object" interface="wl_output"/>
+    </request>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy the manager">
+        All objects created by the manager will still remain valid, until their
+        appropriate destroy request has been called.
+      </description>
+    </request>
+  </interface>
+
+  <interface name="zwlr_output_power_v1" version="1">
+    <description summary="adjust power management mode for an output">
+      This object offers requests to set the power management mode of
+      an output.
+    </description>
+
+    <enum name="mode">
+      <entry name="off" value="0"
+             summary="Output is turned off."/>
+      <entry name="on" value="1"
+             summary="Output is turned on, no power saving"/>
+    </enum>
+
+    <enum name="error">
+      <entry name="invalid_mode" value="1" summary="inexistent power save mode"/>
+    </enum>
+
+    <request name="set_mode">
+      <description summary="Set an outputs power save mode">
+        Set an output's power save mode to the given mode. The mode change
+        is effective immediately. If the output does not support the given
+        mode a failed event is sent.
+      </description>
+      <arg name="mode" type="uint" enum="mode" summary="the power save mode to set"/>
+    </request>
+
+    <event name="mode">
```

Comment:
It occurs to me that this is late in the review process, and I apologise - but this caught my eye and it doesn't really sit well for me. The language here overlaps with *mode*setting. Can we call this "state", perhaps?

Reply:
that would give

```
 <request name="set_state">
      <description summary="Set an outputs power save state">
        Set an output's power save state to the given state. The state change
        is effective immediately. If the output does not support the given
        state a failed event is sent.
      </description>
      <arg name="state" type="uint" enum="state" summary="the power save state to set"/>
    </request>
```

which also reads odd to me. Also dpms talks about `modes`. Maybe someone has an idea besides `mode` or `state`? Otherwise

Reply:
Do you have any documentation from display or GPU vendors which describes this feature? Maybe we can ape their vocabulary, or at least lend some legitimacy to the "mode" conflict.

Reply:
Perhaps "status" would work?

Reply:
I looked at https://en.wikipedia.org/wiki/VESA_Display_Power_Management_Signaling : `DPMS defines four modes; normal, standby, suspended and off` and https://tldp.org/HOWTO/Battery-Powered/displaytypes.html but the wikipedia one is loke base on the tldp one. X besides modes also talks about 'power levels' (which is longer but seems to fit). My order of preference (best to worst is)

1. mode (used elsewhere)
2. power level (used elsewhere but longer)
3. state 
4. status (odd plural form)

I'm not really hooked to any of these though so please just pick one and i'll brush up the protocol.

Reply:
I'm fine with sticking with "mode" because the symbols are prefixed with "power".

Reply:
Alright, mode it is.